Makefile

```diff
@@ -351,8 +351,13 @@ compile_flags.txt: Makefile
 PHONY/clang-tidy-%: compile_flags.txt xlicense.h $(STFL_HDRS) $(NEWSBOATLIB_OUTPUT)
 	$(CLANG_TIDY) $(@:PHONY/clang-tidy-%=%)
 
+PHONY/fix-clang-tidy-%: compile_flags.txt xlicense.h $(STFL_HDRS) $(NEWSBOATLIB_OUTPUT)
+	-$(CLANG_TIDY) --fix $(@:PHONY/fix-clang-tidy-%=%)
```

Comment:
Out of curiosity, why would we ignore the exit status? (I believe that's what the `-` prefix does.)

Reply:
Yes, it does that.

This would otherwise stop at the first issue, even if there is automatic fix applied.
I think it comes from the `.clang-tidy` config `WarningsAsErrors: '*'`, which turns the would-be-warnings to hard errors. That is good for the CI check, but not so suitable for this. For this it's better to let it continue and apply all available fixes.

How I see the workflow with this is:

- add a new clang-tidy check, or after code changes
- run `make fix-clang-tidy` to apply any automatic fixes
- review the changes, and commit or edit some more
- run `make clang-tidy` to check issues left without automatic fix

Reply:
Got it! Make sense :+1: